themes/bootstrap3/templates/RecordDriver/SolrOverdrive/result-list.phtml

```diff
@@ -142,7 +142,7 @@
       <?=$this->record($this->driver)->renderTemplate('controls/qrcode.phtml', ['driver' => $this->driver, 'context' => 'results'])?>
 
       <?php if ($this->cart()->isActiveInSearch() && isset($this->params) && $this->params->getOptions()->supportsCart() && $this->cart()->isActive()): ?>
-        <?=$this->render('record/cart-buttons.phtml', ['id' => $this->driver->getUniqueId(), 'source' => $this->driver->getSourceIdentifier()]); ?><br>
+        <?=$this->render('record/cart-buttons.phtml', ['cartId' => $this->record($this->driver)->getUniqueIdWithSource()]); ?><br>
```

Comment:
If we're going to change the values passed to the cart-buttons.phtml template, maybe it would make the most sense to just pass in the record driver so more granular decisions can be made within that template.

However, changing this part of the code is fairly backward-compatibility-breaking and might not be a good change to make in a minor release; we might want to reschedule this part of the work for release 10.